### PR TITLE
Add missing job selectors to dashboards

### LINKED ISSUE #673: Tighten up prometheus queries to avoid collisions
I was puzzled as to why looking at our dashboards, the resource requests/limits were double what I expected.
Deconstructing the queries, I saw that e.g. https://github.com/kubernetes-monitoring/kubernetes-mixin/blob/7120319a5b5c45e8dd2e79f0ad60e2284c6d6f1b/dashboards/resources/cluster.libsonnet#L164 was unintentionally summing in metrics from kubecost (https://github.com/kubecost/cost-model/blob/068510c25953fc176d85c0ac756f31579f78a67d/pkg/metrics/podmetrics.go#L62)

One possible solution would be to add `job="kube-state-metrics"` to the query.

### PR DESCRIPTION
**Pull request by james-callahan**
Fixes #673

### CODE REVIEW

**Comment by arajkumar** [comment]
@james-callahan Can you please fix the conflicts?

File: dashboards/resources/cluster.libsonnet [hunk 1]
```diff
@@ -181,7 +181,7 @@ local template = grafana.template;
             'sum(namespace_cpu:kube_pod_container_resource_requests:sum{%(clusterLabel)s="$cluster"}) by (namespace)' % $._config,
             'sum(node_namespace_pod_container:container_cpu_usage_seconds_total:sum_irate{%(clusterLabel)s="$cluster"}) by (namespace) / sum(namespace_cpu:kube_pod_container_resource_requests:sum{%(clusterLabel)s="$cluster"}) by (namespace)' % $._config,
             'sum(namespace_cpu:kube_pod_container_resource_limits:sum{%(clusterLabel)s="$cluster"}) by (namespace)' % $._config,
-            'sum(node_namespace_pod_container:container_cpu_usage_seconds_total:sum_irate{%(clusterLabel)s="$cluster"}) by (namespace) / sum(namespace_cpu:kube_pod_container_resource_limits:sum{%(clusterLabel)s="$cluster"}) by (namespace)' % $._config,
+            'sum(node_namespace_pod_container:container_cpu_usage_seconds_total:sum_irate{%(clusterLabel)s="$cluster"}) by (namespace) / sum(namespace_cpu:kube_pod_container_resource_limits:sum{%(kubeStateMetricsSelector)s,%(clusterLabel)s="$cluster"}) by (namespace)' % $._config,
```
[the remark this answers is not in the document]
Comment (james-callahan) [reply dashboards/resources/cluster.libsonnet:184]: That link is to `namespace_cpu:kube_pod_container_resource_limits:sum`, this line here is `node_namespace_pod_container`...

**Comment by james-callahan** [comment]
Fixed and rebased again.

**Comment by povilasv** [comment]
build seems failling

**Comment by james-callahan** [comment]
> build seems failling

Sorry, I messed up the rebase. Fixed now.

**Comment by james-callahan** [comment]
@povilasv please consider for quick merging; rebasing this is error prone :(

**Review by povilasv — APPROVED** [review]
Thanks for the work, I've looked at it again and LGTM.

**Comment by zerkms** [comment]
I think it introduced a regression: https://github.com/kubernetes-monitoring/kubernetes-mixin/issues/697

**Comment by james-callahan** [comment]
> I think it introduced a regression: #697

I don't know how I messed that up

**Comment by james-callahan** [comment]
> I don't know how I messed that up

Ah; I think I got confused because we have `job="kubelet"` for both cadvisor and kubelet.

**Pull request merged**